corehq/apps/reports/static/reports/javascripts/reports.config.js

```diff
@@ -110,12 +110,12 @@ var HQReport = function (options) {
             $.cookie(self.toggleFiltersCookie, showStr, {path: self.urlRoot, expires: 1});
         };
         
-        if ($.cookie(self.toggleFiltersCookie) === null) {
+        if ($.cookie(self.toggleFiltersCookie) === undefined || $.cookie(self.toggleFiltersCookie) === null) {
```

Comment:
Thanks Daniel. So the funny part is that when the report filters are not supposed to be shown, the cookie is set to an empty string.

How about:
```javascript
    if (!$.cookie(self.toggleFiltersCookie) && 
        $.cookie(self.toggleFiltersCookie) !== '') {
```
Still a bit clunky, but works.